packages/deltas-websocket/src/client/client-impl.ts

```diff
@@ -168,33 +167,82 @@ export class LionWebClient {
-            return this.lowLevelClient.sendMessage(queryRequest)
+            this.queryResolveById[queryRequest.queryId] = resolveResponse
+            this.lowLevelClient.sendMessage(queryRequest)
+                .catch(rejectResponse)
         })
 
-    signOn = async (queryId: LionWebId) =>
+    async subscribeToChangingPartitions(queryId: LionWebId, parameters: SubscribeToPartitionChangesParameters): Promise<void> {
         await this.enqueueQuery({
+            messageKind: "SubscribeToChangingPartitionsRequest",
+            queryId,
+            ...parameters,
+            protocolMessages: []
+        } as SubscribeToChangingPartitionsRequest)
+    }
+
+    async subscribeToPartitionContents(queryId: LionWebId, partition: LionWebId): Promise<LionWebJsonChunk> {   // TODO  already deserialize, because we've got everything we need
+        const response = await this.enqueueQuery({
+            messageKind: "SubscribeToPartitionContentsRequest",
+            queryId,
+            partition,
+            protocolMessages: []
+        } as SubscribeToPartitionContentsRequest) as SubscribeToPartitionContentsResponse
+        return response.contents
+    }
+
+    async unsubscribeFromPartitionContents(queryId: LionWebId, partition: LionWebId): Promise<void> {
+        await this.enqueueQuery({
+            messageKind: "UnsubscribeFromPartitionContentsRequest",
+            queryId,
+            partition,
+            protocolMessages: []
+        } as UnsubscribeFromPartitionContentsRequest)
+    }
+
+    async signOn(queryId: LionWebId): Promise<void> {
+        if (this.signedOff) {
+            return Promise.reject(new Error(`can't sign on after having signed off`))
+        }
+        const response = await this.enqueueQuery({
             messageKind: "SignOnRequest",
             queryId,
             deltaProtocolVersion: "2025.1",
             clientId: this.clientId,
             protocolMessages: []
-        } as SignOnQueryRequest)
+        } as SignOnRequest) as SignOnResponse
+        this._participationId = response.participationId
+    }
 
-    signOff = async (queryId: LionWebId) =>
+    async signOff(queryId: LionWebId): Promise<void> {
         await this.enqueueQuery({
             messageKind: "SignOffRequest",
             queryId,
             protocolMessages: []
-        } as SignOffQueryRequest)
+        } as SignOffRequest)
+        this.signedOff = true
+        this._participationId = undefined
+    }
 
-    async disconnect() {
+    async reconnect(queryId: LionWebId, participationId: LionWebId, lastReceivedSequenceNumber: number): Promise<void> {
+        const response = await this.enqueueQuery({
+            messageKind: "ReconnectRequest",
+            queryId,
+            participationId,
+            lastReceivedSequenceNumber,
+            protocolMessages: []
+        } as ReconnectRequest) as ReconnectResponse
```

Comment:
depending how much error handling we're already considering, this might be an error event telling the participation is outdated.

Reply:
It is mentioned in the error section there.